### PR TITLE
[testing] Enable automatic integration tests for linstor module

### PR DESCRIPTION
**Pull request by kvaps**
## Description

This PR enables helm test for linstor module.

This is temporary solution before implementing https://github.com/deckhouse/deckhouse/issues/2380

## Why do we need it, and what problem does it solve?

Execute e2e tests for linstor

## What is the expected result?

```changes
section: testing
type: chore
summary: Enable automatic integration tests for linstor module
impact_level: default
```

### CODE REVIEW

**Comment by kvaps** [comment]
e2e tests passed as part of https://github.com/deckhouse/deckhouse/pull/2359#issuecomment-1235162020

**Review by diafour — DISMISSED** [review]

**Review by yalosev — DISMISSED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @nabokihms. I've started the workflow for label 'e2e/run/static'...

**Review by nabokihms — CHANGES_REQUESTED** [review]
We need to figure out what is wrong with tests.

**Comment by kvaps** [comment]
@nabokihms what do you mean, helm tests are executed after `wait_cluster_ready` function

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @kvaps. I've started the workflow for label 'e2e/run/static'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @kvaps. I've started the workflow for label 'e2e/run/static'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @kvaps. I've started the workflow for label 'e2e/run/static'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @kvaps. I've started the workflow for label 'e2e/run/static'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @kvaps. I've started the workflow for label 'e2e/run/static'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @kvaps. I've started the workflow for label 'e2e/run/static'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @kvaps. I've started the workflow for label 'e2e/run/static'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @kvaps. I've started the workflow for label 'e2e/run/static'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @kvaps. I've started the workflow for label 'e2e/run/static'...

**Review by nabokihms — CHANGES_REQUESTED** [review]
The output of tests is a single string. It seems unclear to me what tests were run and their statuses.
https://github.com/deckhouse/deckhouse/actions/runs/3099855694/jobs/5019492304#step:12:2547

1. Add the header that this is the Linstore module test.
2. Add verbosity. I expect to see something like this.
https://github.com/dexidp/helm-charts/actions/runs/3065654816/jobs/4950225382

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @kvaps. I've started the workflow for label 'e2e/run/static'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @kvaps. I've started the workflow for label 'e2e/run/static'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @kvaps. I've started the workflow for label 'e2e/run/static'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @kvaps. I've started the workflow for label 'e2e/run/static'...

**Comment by kvaps** [comment]
> Done

```
Running linstor suit tests ...
+ kubectl -n d8-system exec deploy/deckhouse -- helm test -n d8-system linstor
NAME: linstor
LAST DEPLOYED: Thu Sep 29 22:32:10 2022
NAMESPACE: d8-system
STATUS: deployed
REVISION: 3
TEST SUITE:     linstor-test-storage-class
Last Started:   Thu Sep 29 22:46:27 2022
Last Completed: Thu Sep 29 22:46:27 2022
Phase:          Succeeded
TEST SUITE:     linstor-test-wait-api
Last Started:   Thu Sep 29 22:46:23 2022
Last Completed: Thu Sep 29 22:46:24 2022
Phase:          Succeeded
TEST SUITE:     linstor-test-create-storage-pool
Last Started:   Thu Sep 29 22:46:24 2022
Last Completed: Thu Sep 29 22:46:27 2022
Phase:          Succeeded
TEST SUITE:     linstor-test-pod-with-volume
Last Started:   Thu Sep 29 22:46:28 2022
Last Completed: Thu Sep 29 22:46:41 2022
Phase:          Succeeded
TEST SUITE:     linstor-test-pools-importer
Last Started:   Thu Sep 29 22:46:41 2022
Last Completed: Thu Sep 29 22:46:57 2022
Phase:          Succeeded
TEST SUITE:     linstor-test-cleanup
Last Started:   Thu Sep 29 22:46:57 2022
Last Completed: Thu Sep 29 22:47:00 2022
Phase:          Succeeded
E2E test: Success!
```

https://github.com/deckhouse/deckhouse/actions/runs/3154940116

File: testing/cloud_layouts/script.sh [hunk 1]
```diff
@@ -613,6 +617,44 @@ ENDSSH
...
+#  - ssh_user
+#  - master_ip
+#
+# TODO: replace with testing framework: https://github.com/deckhouse/deckhouse/issues/2380
+function run_linstor_tests() {
+  test_failed=
+
+  testScript=$(cat <<"END_SCRIPT"
+set -Eeuo pipefail
+>&2 echo "Running linstor suit tests ..."
```
Comment (diafour) [inline testing/cloud_layouts/script.sh:633]:
```suggestion
>&2 echo "Running linstor test suite ..."
```

File: testing/cloud_layouts/script.sh [hunk 2]
```diff
@@ -613,6 +617,44 @@ ENDSSH
...
+set -x
+>&2 kubectl -n d8-system exec deploy/deckhouse -- helm test -n d8-system linstor
+END_SCRIPT
+)
+
+  testRunAttempts=5
+  for ((i=1; i<=$testRunAttempts; i++)); do
+    if ssh -o GlobalKnownHostsFile=/dev/null -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -i "$ssh_private_key_path" "$ssh_user@$master_ip" sudo -i /bin/bash <<<"${testScript}"; then
```
Comment (diafour) [inline testing/cloud_layouts/script.sh:641]: There is a variable `$ssh_command` to run ssh after #2283

Comment (kvaps) [reply testing/cloud_layouts/script.sh]: done

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @kvaps. I've started the workflow for label 'e2e/run/static'...

**Comment by kvaps** [comment]
@nabokihms could you review this please?

File: testing/cloud_layouts/script.sh [hunk 3]
```diff
@@ -322,6 +322,10 @@ function run-test() {
     change_deckhouse_image "${SWITCH_TO_IMAGE_TAG}" || return $?
     wait_cluster_ready || return $?
   fi
+
+  if [[ "$PROVIDER" == "Static" ]]; then
+    run_linstor_tests || return $?
+  fi
```
Comment (RomanenkoDenys) [inline testing/cloud_layouts/script.sh:328]: Please move this peace of code inside `wait_cluster_ready` function before upmeter probes.

**Review by RomanenkoDenys — CHANGES_REQUESTED** [review]
one small change

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @kvaps. I've started the workflow for label 'e2e/run/static'...

**Review by diafour — DISMISSED** [review]

File: testing/cloud_layouts/script.sh [hunk 4]
```diff
@@ -613,6 +617,44 @@ ENDSSH
...
+    else
+      test_failed="true"
+      >&2 echo "Run test script via SSH: attempt $i/$testRunAttempts failed. Sleeping 30 seconds..."
+      sleep 30
+    fi
+  done
+
+  if [[ $test_failed == "true" ]] ; then
+    return 1
+  fi
```
Comment (diafour) [inline testing/cloud_layouts/script.sh:653]:
```suggestion
  fi
  echo "Linstor test suite: success"
```

Let's indicate success explicitly.

File: testing/cloud_layouts/script.sh (hunk 3 above)
Comment (kvaps) [reply testing/cloud_layouts/script.sh]: done

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @kvaps. I've started the workflow for label 'e2e/run/static'...

**Review by nabokihms — APPROVED** [review]

**Pull request merged**